app/src/pages/UserSelectors.js

```
@@ -1,21 +1,10 @@
 export const getBalance = state => {
-  const ethAccounts = state.user.data.ethAccounts
-  if (ethAccounts) {
-    return state.user.data.ethAccounts.length > 0
-      ? state.user.data.ethAccounts[0].sanBalance
-      : 0
-  }
-  return 0
+  return state.user.data.sanBalance > 0
+    ? state.user.data.sanBalance : 0
 }
 
 export const checkHasPremium = state => {
-  const ethAccounts = state.user.data.ethAccounts
-  if (ethAccounts) {
-    return state.user.data.ethAccounts.length > 0
-      ? state.user.data.ethAccounts[0].sanBalance >= 1000
-      : false
-  }
-  return false
+  return state.user.data.sanBalance >= 1000
```

Not good to be hardcoded IMO. If the user has not enough SAN tokens, the GQL query returns `{:error, "Insufficient SAN balance"}`. Can we use it somehow?

---

I share your concerns but I think this is out of the scope of this PR cause it will require quite a refactoring in the frontend. The goal here is to use the `san_balance` from the `currentUser` and to simplify these functions by using it. Also the hardcoded value gives flexibility in the tooltip/error message and saves one backend call in the price to not use the backend as single source of truth.

---

Seems like this hardcoded value is here before this PR. @Partysun do you have ideas how we can improve that? I would suggest to schedule a task to make this amount easily configurable.

---

Added this card to the frontend board: https://favro.com/organization/bdbb4f24afc48b29c74f4fa4/9ff267872fcd8b7925f1d7c7?card=San-1764